### PR TITLE
HTML newsletter template for subscribers

### PR DESCRIPTION
**Pull request by aileen**
refs #7023

The template can be used for all cases (different newsletter interval, different amount of blog posts, with or without blog picture).

The template can be filled with the following data structure:
- `blog.logo` is the blog logo from settings
- `blog.title` is the title of the current blog
- `blog.url` is the URL of the blog
- `blog.twitterURL` is the twitter profile URL of the blog from settings
- `blog.facebookURL` is the facebook page URL of the blog from settings
- `blog.unsubscribe` is the link for the reader to unsubscribe from the blog
- `newsletter.interval` is the interval in words like 'weekly', 'daily', 'monthly'
- `newsletter.date` is the date of the newsletter issue in a format like 'June 9th, 2016'
- `blog.post` is expected to be an array:
  - `blog.post[i].picture` is the picture of the blog post. There are also conditionals, which change the inline CSS to either show a border-top if there's no picture (in that case the HTML code, that shows the picture is not active)
  - `blog.post[i].title` is the title of the current blog post
  - `blog.post[i].text` is the text of the current blog post, which needs to be cut down to 278 letters, plus ending with `&#x2026;`
  - `blog.post[i].url` is the URL of the current blog post
  - `blog.post[i].tag` is the tag of the current blog post
  - `blog.post[i].author` is the author of the current blog post

Important is, that only HEX HTML entities will work, especially in Outlook. So instead of `&mdash;` we need to use `&#x2014;` and `&#x2019;` instead of `&apos;` or `'` and so on.

Added unit test for newsletter template in `test/unit/mail/utils_spec.js`.

---

Some screenshots (there's always one version with a blog picture and one without. I didn't upload all of the examples)

4 or more posts:
newsletter_4_pic

3 posts:
newsletter_3

2 posts:
newsletter_2_pic

1 post:
newsletter_1

### CODE REVIEW

**Comment by kirrg001** [comment]
Could you please add one single unit test?
To ensure that the template itself can be compiled with data.

see https://github.com/TryGhost/Ghost/blob/master/core/test/unit/mail/utils_spec.js

Thank you :)

File: core/test/unit/mail/utils_spec.js [hunk 1]
```diff
@@ -42,4 +42,75 @@ describe('Mail: Utils', function () {
...
+                            tag: 'fourth',
+                            author: 'donald duck'
+                        },
+                        {
+                            picture: 'http://myblog.com/content/images/post-5-image.jpg',
+                            title: 'Fifth blog post',
+                            text: 'This is the fifth blog post. It&#x2019;s also awesome&#x2026;',
+                            url: 'http://myblog.com/fifth-blog-post',
+                            tag: 'fifth',
+                            author: 'casper the ghost'
```
Comment (kirrg001) [inline core/test/unit/mail/utils_spec.js]: hehe i like your authors

**Comment by kirrg001** [comment]
LGTM 👍

**Comment by rawsh** [comment]
Are you planning on including the share links from the blog settings?

**Comment by AileenCGN** [comment]
Hi @rawsh, yes that's the plan
> blog.twitterURL is the twitter profile URL of the blog from settings
> blog.facebookURL is the facebook page URL of the blog from settings

😃

**Pull request merged**